### PR TITLE
Update README, add tests

### PR DESCRIPTION
**Pull request by MartinKolarik**

### CODE REVIEW

**Comment by megawac** [comment]
:clap:

**Comment by jimaek** [comment]
Its jsDelivr, not JSDelivr  :)
Brand awareness and everything

File: README.md [hunk 1]
```diff
@@ -1,72 +1,192 @@
...
-* `name` - name of library. Example: jquery
-* `mainfile` - mainfile parameter in info.ini. Example: jquery.min.js
-* `lastversion`- lastversion of the project. Example: 2.0.3 (will match multiple projects)
-* `versions` -  all hosted versions for selected project. (read only)
-* `description` - description of the project
-* `homepage`- webpage of project. Example: http://jquery.com/
-* `github`- github page of project. Example: https://github.com/jquery/jquery
-* `author` - the author of project. Example: jQuery Foundation
-* `assets` - files hosted per versions. (read only)
+Where `<cdn> == jsdelivr | cdnjs | google | bootstrap`,
```
Comment (jimaek) [inline README.md]: probably need to add jquery here as well

**Comment by jimaek** [comment]
I believe its missing this functionality https://github.com/jsdelivr/api/issues/109 ?

**Comment by MartinKolarik** [comment]
@jimaek yes and yes. Will add that.

File: README.md [hunk 2]
```diff
@@ -1,72 +1,192 @@
...
+`<library>` is a library provided by `<cdn>`
+and `<version>` is some version of `<library>`.
 
+- `/<cdn>/libraries`
+  - Returns an array of library objects for all libraries hosted by <cdn>;
```
Comment (jimaek) [inline README.md]: `<cdn>` is not displayed, you are missing ``

**Comment by jimaek** [comment]
Also its a good idea to add an "Examples" section to help people understand how it works

File: README.md (hunk 2 above)
Comment (MartinKolarik) [inline README.md]: Well I just restored the original content (it got changed to v1 when I merged in some changes from master). But I think it actually displays correctly: https://github.com/MartinKolarik/api/tree/v2

Comment (jimaek) [inline README.md]: Its not, its empty https://dl.dropboxusercontent.com/s/zqh9wziki7exhpz/firefox_2016-01-21_23-25-25.png?dl=0

Comment (MartinKolarik) [inline README.md]: ok, I was looking at the wrong line :)

**Comment by tomByrer** [comment]
> "Examples" section

I noticed it was removed also, but I like the shorter README.  Perhaps more clear `/tests` (API is meant for devs)?  Or separate file?

**Comment by MartinKolarik** [comment]
Updated. I also added tests for the documented stuff.

File: README.md [hunk 3]
```diff
@@ -1,72 +1,181 @@
...
-```
-http://api.jsdelivr.com/v1/jsdelivr/libraries?name=jquery&fields=assets
-```
+#### `library`
 
-Get hosted files for a selected version
-```
-http://api.jsdelivr.com/v1/jsdelivr/libraries/jquery/2.0.3
-```
+Queries to `/v2/<cdn>/libraries` are formatted as follows.
```
Comment (jimaek) [inline README.md]: The header says `library` but you explain the `libraries` endpoint. This is not correct I think

Comment (MartinKolarik) [inline README.md]: It is. This section explains what is a `library` object and `error` object, i.e. what properties they have, and that applies to both/all endpoints. A list of endpoints is below.

**Pull request merged**